File: js/que_.js
'use strict';

let DLL = require('./dll')

class Queue{
    constructor(){
        this.dll = new DLL()
    }

    enqueue(val){
        this.dll.push(val)
    }

    dequeue(){
        try{
            return this.dll.shift();
        }catch(e){
            return undefined;
        }
    }

    peek(){
        if(this.dll._counter === 0){
            return undefined
        }
        return this.dll.tail.data;
    }

    size(){
        return this.dll._counter;
    }
}

module.exports = Queue;

File: src/que_.py
"""Make a Queue object that has access to DoublyLinkedList methods."""
from dll import DLL


class Queue(object):
    """Create a Queue object."""

    def __init__(self):
        """Initialize Queue using DLL methods."""
        self._dll = DLL()

    def enqueue(self, val):
        """Add a node to queue."""
        self._dll.push(val)

    def dequeue(self):
        """Remove first node pushed in."""
        try:
            return self._dll.shift()
        except IndexError:
            raise IndexError('Nothing to dequeue')